velox/functions/prestosql/tests/JsonFunctionsTest.cpp

```diff
@@ -67,6 +74,171 @@ TEST_F(JsonFunctionsTest, jsonArrayLength) {
+      false);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(R"([false, false, false])", 17), false);
+  EXPECT_EQ(json_array_contains<int64_t>(R"(1)", 1), std::nullopt);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(R"("thefoxjumpedoverthefence")", 1),
+      std::nullopt);
+  EXPECT_EQ(json_array_contains<int64_t>(R"("")", 1), std::nullopt);
+  EXPECT_EQ(json_array_contains<int64_t>(R"(true)", 1), std::nullopt);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(R"({"k1":[0,1,2], "k2":"v1"})", 1),
+      std::nullopt);
+
+  EXPECT_EQ(json_array_contains<int64_t>(R"([1, 2, 3])", 1), true);
+  EXPECT_EQ(json_array_contains<int64_t>(R"([1, 2, 3])", 4), false);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(R"([123, 123.456, true, "abc"])", 123),
+      true);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(R"([123, 123.456, true, "abc"])", 456),
+      false);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(
+          R"([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20])",
+          17),
+      true);
+  EXPECT_EQ(
+      json_array_contains<int64_t>(
+          R"([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20])",
+          23),
+      false);
+}
+
+TEST_F(JsonFunctionsTest, jsonArrayContainsDouble) {
+  EXPECT_EQ(json_array_contains<double>(R"([])", 2.3), false);
+  EXPECT_EQ(json_array_contains<double>(R"([1, 2, 3])", 2.3), false);
+  EXPECT_EQ(
+      json_array_contains<double>(R"(["hello", "presto", "world"])", 2.3),
+      false);
+  EXPECT_EQ(
+      json_array_contains<double>(R"([false, false, false])", 2.3), false);
+  EXPECT_EQ(json_array_contains<double>(R"(1)", 2.3), std::nullopt);
+  EXPECT_EQ(
+      json_array_contains<double>(R"("thefoxjumpedoverthefence")", 2.3),
+      std::nullopt);
+  EXPECT_EQ(json_array_contains<double>(R"("")", 2.3), std::nullopt);
+  EXPECT_EQ(json_array_contains<double>(R"(true)", 2.3), std::nullopt);
+  EXPECT_EQ(
+      json_array_contains<double>(R"({"k1":[0,1,2], "k2":"v1"})", 2.3),
+      std::nullopt);
+
+  EXPECT_EQ(json_array_contains<double>(R"([1.2, 2.3, 3.4])", 2.3), true);
+  EXPECT_EQ(json_array_contains<double>(R"([1.2, 2.3, 3.4])", 2.4), false);
+  EXPECT_EQ(
+      json_array_contains<double>(R"([123, 123.456, true, "abc"])", 123.456),
+      true);
+  EXPECT_EQ(
+      json_array_contains<double>(R"([123, 123.456, true, "abc"])", 456.789),
+      false);
+  EXPECT_EQ(
+      json_array_contains<double>(
+          R"([1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5])",
+          4.5),
+      true);
+  EXPECT_EQ(
+      json_array_contains<double>(
+          R"([1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5, 1.2, 2.3, 3.4, 4.5])",
+          4.3),
+      false);
+}
+
+TEST_F(JsonFunctionsTest, jsonArrayContainsString) {
+  EXPECT_EQ(json_array_contains<std::string>(R"([])", ""), false);
```

Comment:
Please add some test cases involving large strings : both in the json array and the value being searched for.